### PR TITLE
Add progress bar to loader-tests

### LINKED ISSUE #143: Add progress to Check OPML script
We have `loader-tests/check-opml.ts` script to check feeds from OPML file. You can run it by:

```sh
pnpm check-opml ./loader-tests/example.ompl
```

We want to add progress bar to the process like:

```
███████████░░░░░
```

In `loader-tests` we plan to have many scripts, so this progress bar should re-usable from `loader-tests/utils.ts`. For instance:
1. In the start script define jobs number by `setJobs(number)`
2. On every `success()`/`error()` call we mark next job as completed.

We should not show progress on CI.

### PR DESCRIPTION
**Pull request by Kamino0**
Fixes #143
Added progress bar to loader-tests.

Also changed `&` to `&amp;` in example.opml to satisfy parser.
Recently there was a [fix](https://github.com/hplush/slowreader/commit/ef1cea249e5e4d50198167c41fac90a1f426fe99) to the сontentType definition and on my computer [queryselectorAll](https://github.com/hplush/slowreader/blob/main/loader-tests/check-opml.ts#L22) started returning an empty list.
I think the problem in more strict parsing of XML in comparison with HTML, since previously the example.opml file was defined as text/html and now as application/xml.
I suppose in future resolving issue #195 fix this problem.

progress-bar

# Checklist

- [x] Don’t rush. Check all changes in PR again.
- [x] Run `pnpm test`.
- [x] Think about changing documentation.
  - If you added a script to `scripts/`, add a comment with a description.
  - If you added a new folder, add its description to the project’s `README.md`.
  - If you added config, describe how we use this tool in the config’s comment.
  - If you added something to the project’s architecture, describe it in the project’s `README.md`.
  - Try to focus on “why?”, not “how?”.
- [x] If you added a new dependency, check our [requirements](https://github.com/hplush/slowreader/blob/main/README.md#dependencies).
- [x] Think about testing
  - If you added a feature, add unit tests.
  - If you added a new state to the UI, add visual tests.
  - If you fixed the bug, think about preventing bug regression in the future.
- If you changed web client:
  - [ ] Think about moving code to `core/`. What code will also be useful on other platforms?
  - [ ] Run `pnpm size` and check the difference in the JS bundle size. Is it relevant to the changes? Change the limit in `web/.size-limit.json` if necessary.
  - [ ] The UI was checked in Chrome and Firefox (and Safari or Epiphany if you have them).
  - [ ] Think about keyboard UX. Is it easy to use the new feature with only one hand on a keyboard? Is it easy to understand what keys to press?
  - [ ] Think about HTML semantics.
  - [ ] Think about accessibility. Check a11y recommendations. Think about how screen reader users will use the tool. Is it easy to use on a screen with bad contrast?
  - [ ] Think about translations.
  - [ ] Think about right-to-left languages. What parts of the screen should be mirrored for Arabic or Hebrew languages?
- If you changed `web/main/` files:
  - Do you need to backport styles changes to `web/public/404.html` or `web/public/500.html`?
  - Think about app loading styles inlined in `index.html`.
- If you changed `core/`:
  - [ ] Think about making types more precise. Can you better explain data relations by type?
  - [ ] Think about conflict resolution. We don’t need some very smart changing merging; just 2 changes of the same item on different clients should not break the database. What if the user changes an item on one machine and removes it on another?
  - [ ] Think about log and storage migration.
- If you changed English translations:
  - [ ] Change translation ID if you change the meaning of the text.

### CODE REVIEW

**Comment by ai** [comment]
Can you show screenshot of progress bar?

File: loader-tests/check-opml.ts [hunk 1]
```diff
@@ -59,6 +60,10 @@ cli.run(async args => {
   }
 
   let feeds = await parseFeedsFromFile(opmlFile)
+
+  const jobsCount = home ? feeds.length * 2 : feeds.length
```
Comment (ai) [inline loader-tests/check-opml.ts:64]: Run `eslint --fix .` to fix code style.

And you marked `Run pnpm test` as done, but didn’t call it ;).

File: loader-tests/check-opml.ts [hunk 2]
```diff
@@ -10,7 +10,8 @@ import {
   finish,
```
Comment (ai) [inline loader-tests/check-opml.ts:10]: We need progress bar also in `test-loaders`

**Comment by Kamino0** [comment]
> Can you show screenshot of progress bar?

Added gif to main comment

File: loader-tests/check-opml.ts (hunk 2 above)
Comment (Kamino0) [reply loader-tests/check-opml.ts:10]: done

File: loader-tests/check-opml.ts (hunk 1 above)
Comment (Kamino0) [reply loader-tests/check-opml.ts]: fixed 👀

Comment (Kamino0) [reply loader-tests/check-opml.ts]: I called, but it seems these errors were hidden under a lot of 
```
Unable to read file core/coverage/tmp/coverage-49309-1715592007988-0.json
ENOENT: no such file or directory, open /core/coverage/tmp/coverage-49309-1715592007988-0.json
```

Want look at this errors later

**Comment by ai** [comment]
Thanks! Great job.

**Pull request merged**